examples/advanced/maximal_propagate.py

```diff
@@ -0,0 +1,39 @@
+"""
+Example showing repeated solving for deriving the maximal consequence of set of constraints.
+
+Iteratively finds a new solutions by forcing at least one variable to have a value not seen in any other solution.
+"""
+
+from cpmpy import *
+from cpmpy.transformations.get_variables import get_variables
+
+
+def maximal_propagate(constraints, solvername="ortools"):
```

Comment:
Proposed improvements:
- Users can choose to use their own solver or start a new one.
- Users can choose which variables to propagate (i.e. explanations use-case).

```python
def maximal_propagate(constraints, vars=None, solver=None, solvername="ortools"):
```

Reply:
What would be the usecase for providing your own solver? Add some extra hard constraints or?
Providing the vars might be a good idea indeed, this will also limit the number of iterations needed

Reply:
Providing your own solver can improve efficiency (in case of incremental solving) but then you also have to have assumption variables to not make the solver contain inconsistent constraints. It's ok to keep things simple and go for the advanced use case when the need arises.